nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -205,9 +205,10 @@ def _build_tokenizer(self):
         self.tokenizer = get_nmt_tokenizer(
             library=self._cfg.tokenizer.library,
             model_name=self._cfg.tokenizer.type,
-            tokenizer_model=self.register_artifact("tokenizer.model", self._cfg.tokenizer.model),
-            vocab_file=self.register_artifact("tokenizer.vocab_file", self._cfg.tokenizer.vocab_file),
-            merges_file=self.register_artifact("tokenizer.merge_file", self._cfg.tokenizer.merge_file),
+            tokenizer_model=self.register_artifact("tokenizer.model", self._cfg.tokenizer.get('model', None)),
```

Comment:
@trias702 could you please comment on that

Reply:
@MaximumEntropy It happens if you use a HF tokenizer. To use a pretrained HF tokenizer like `EleutherAI/gpt-neox-20b` or `bert-base-uncased` you don't need to pass any `model`, `vocab_file`, or `merge_file` in the config, you just need to pass `tokenizer.library=huggingface` and `tokenizer.type=EleutherAI/gpt-neox-20b`. But because the old code would call register_artifact for `cfg.tokenizer.model` which would be None, it would throw an error. That's why I changed this block, to ensure it works correctly with pure HF tokenizers at model instantiation.